File: kotlin/java-basic-example-vs-kotlin/src/main/java/org/example/ex07/Exam07.java
package org.example.ex07;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author : Rene
 * @since : 2023/10/22
 */
public class Exam07 {
}



// 롬복을 사용하면 애노테이션으로 작성하지만 , 실제로는 보일러플레이트 코드를 전부 작성해야 함
@Data
@NoArgsConstructor
@AllArgsConstructor
class User{
	private String name;
	private String email;
	private int age;
}

File: kotlin/kotlin-basic-example/src/main/kotlin/ex07/Exam07.kt
package ex07

/**
 * @author        : Rene
 * @since         : 2023/10/22
 */
class User(
    var name: String? = null, var age: Int? = null, var eamil: String? = null
) {
    override fun toString(): String {
        return "User(name=$name, age=$age, eamil=$eamil)"
    }
}

fun main() {
    var user = User()
    user.name= "홍길동"
}